dotnet/src/Connectors/Connectors.UnitTests/OpenAI/KernelConfigOpenAIExtensionsTests.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq;
+using Microsoft.SemanticKernel;
+using Moq;
+using Xunit;
+
+namespace SemanticKernel.Connectors.UnitTests.OpenAI;
+
+/// <summary>
+/// Unit tests of <see cref="KernelConfigOpenAIExtensions"/>.
+/// </summary>
+public class KernelConfigOpenAIExtensionsTests
+{
+    [Fact]
+    public void ItFailsWhenAddingTextCompletionServicesWithSameId()
+    {
+        var target = new KernelConfig();
+        target.AddAzureOpenAITextCompletionService("azure", "depl", "https://url", "key");
+
+        var exception = Assert.Throws<KernelException>(() =>
+        {
+            target.AddAzureOpenAITextCompletionService("azure", "depl2", "https://url", "key");
+        });
+        Assert.Equal(KernelException.ErrorCodes.InvalidServiceConfiguration, exception.ErrorCode);
```

Comment:
Nit: slightly confusing name - would prefer "DuplicateServiceConfiguration", but it seems to only be used for repeated service IDs, so I'm not going to hang the PR on a rename.

Reply:
we'll leave it for now, since this code is going to be revisited at least twice more before v1